src/WebJobs.Extensions.DurableTask/ContextInterfaces/DurableContextExtensions.cs

```diff
@@ -0,0 +1,402 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Defines convenient overloads for calling the context methods, for all the contexts.
```

Comment:
This is an interesting approach for extending interfaces. It's a little unfortunate that it has to be in a separate class since that makes it harder for users to find these methods when browsing the API documentation. I wonder if [C# 8's Default Interface Implementation](https://www.codejourney.net/2019/02/csharp-8-default-interface-methods/) can help with this. If so, I think we should consider this as an alternative approach (of course, this would require us to take a dependency on the newer tools which are not yet GA).

Reply:
Which API browsing experience are you referring to?
-  I tested the normal Intellisense, it works great (i.e. I get the full API list, including the extension methods, when typing a dot after a context-typed expression in VS).  
- Inspection in the object browser in VS is definitely not nice, not just because of the extension class, but also because of the many layers of interfaces.
- Not sure if there are any automatically generated API documentations.

C# 8 default interfaces would definitely be great for this.